### PR TITLE
Hardcoded line separator bug

### LINKED ISSUE #552: Hardcoded line separator bug
The bug occurs when the Javadoc text formed with Windows `\r\n` line separator.

For example, we have the comment `A\r\n\r\nB\r\n`. Because `JavaPoet` uses hardcoded line separator, `CodeWriter#emitAndIndent(String)` splits the comment on the next lines: `A\r`, `\r`, `B\r`. As the second line is not empty, indentation emitted.

It's a problem, if someone wants to test generation of Javadoc, that uses `JavaPoet`. Generated Javadoc depends on a platform (Windows or Unix). On Unix the Javadoc empty lines are really empty, but on Windows they are indented. In reason of this unreal to test generation of a Javadoc on different platforms.

Hardcoded line separator may cause and others bugs also.

### PR DESCRIPTION
**Pull request by topalavlad**
Use regex for new line character to cover both dos and unix endings when calling emitAndIndent - fixes #552

### CODE REVIEW

File: src/main/java/com/squareup/javapoet/CodeWriter.java [hunk 1]
```diff
@@ -456,7 +456,7 @@ private ClassName stackClassName(int stackDepth, String simpleName) {
    */
   CodeWriter emitAndIndent(String s) throws IOException {
     boolean first = true;
-    for (String line : s.split("\n", -1)) {
+    for (String line : s.split("\r?\n", -1)) {
```
Comment (h908714124) [inline src/main/java/com/squareup/javapoet/CodeWriter.java]: Javapoet is a Java 8 project, so we might be able to use the linebreak matcher `\\R` instead.

Comment (topalavlad) [reply src/main/java/com/squareup/javapoet/CodeWriter.java]: I'll update the PR. Thanks

**Comment by Egorand** [comment]
@topalavlad I'm going to merge this (sorry it took a decade!) but we'll need you to sign our [CLA](https://spreadsheets.google.com/spreadsheet/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) first. Please let me know when it's done so I can proceed. Thanks!

**Comment by topalavlad** [comment]
> @topalavlad I'm going to merge this (sorry it took a decade!) but we'll need you to sign our [CLA](https://spreadsheets.google.com/spreadsheet/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1) first. Please let me know when it's done so I can proceed. Thanks!

Signed it. Happy new year!

**Comment by Egorand** [comment]
Awesome, thanks! Happy New Year to you as well!

**Pull request merged**